double-conversion/strtod.cc

```diff
@@ -483,7 +483,27 @@ float Strtof(Vector<const char> buffer, int exponent) {
   double double_guess;
   bool is_correct = ComputeGuess(trimmed, exponent, &double_guess);
 
-  float float_guess = static_cast<float>(double_guess);
+  // ASAN has a sanitize check that disallows casting doubles to floats if
+  // they are too big.
+  // https://clang.llvm.org/docs/UndefinedBehaviorSanitizer.html#available-checks
+  // The behavior should be covered by IEEE 754, but some projects use this
+  // flag, so work around it.
+  float max_finite = 3.4028234663852885981170418348451692544e+38;
+  // The half-way point between the max-finite and infinity value.
+  // Since infinity has an even significand everything equal or greater than
+  // this value should become infinity.
+  double half_max_finite_infinity =
+      3.40282356779733661637539395458142568448e+38;
```

Comment:
Where is this value from? (how is 'halfway' between float_max and infinity defined? i'm not too familiar with the IEEE standard)

Reply:
The easiest way to compute it is to take the difference between the max_finite and the previous single, and to add half of that to max_finite.

Max_finite: 3.4028234663852885981170418348451692544e+38
Previous_float: 3.40282326356119256160033759537265639424e+38
max_finite + (max_finite - previous_float)/2: 3.40282356779733661637539395458142568448e+38